woocommerce/woocommerce-android
# Stats Improvements Part I: Add changes to fetch stats data from the v4 API

Fixes the first step of #1199. To keep the PRs smaller, I'm breaking them down for review to be merged into a feature branch. This PR covers the following changes:

- [x] Use the new stats v4 endpoints to fetch data.
- [x] Display the new stats for Today, This Week, This Month, and This Year.
- [x] Consolidate the chart and top selling products to display and update under the same date range.
- [x] Anchor the tabLayout when the view is scrolled

### Notes:
- This [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1302) needs to be merged before this PR can be merged.
- The Unfulfilled Order card is removed in another [PR](https://github.com/woocommerce/woocommerce-android/pull/1245) which is in review.

### Behaviour:
<img width="300" src="https://user-images.githubusercontent.com/22608780/61204692-8a671100-a70b-11e9-96cb-f7b1bd6aee2a.gif">

#### Screenshots:
<img width="300" src="https://user-images.githubusercontent.com/22608780/61204743-b2ef0b00-a70b-11e9-910d-0ee359e8b20a.png"> .  <img width="300" src="https://user-images.githubusercontent.com/22608780/61204760-bc787300-a70b-11e9-8025-eef708490fcf.png"> .  <img width="300" src="https://user-images.githubusercontent.com/22608780/61204781-c8fccb80-a70b-11e9-93ba-dc9211401201.png"> .  <img width="300" src="https://user-images.githubusercontent.com/22608780/61204800-d44ff700-a70b-11e9-9cce-52e6a441115c.png">

#### When no stats data:
<img width="300" src="https://user-images.githubusercontent.com/22608780/61214760-65cc6280-a726-11e9-98b0-6e52b2d55b17.png">

### Testing:
- Run `DashboardPresenterTest`.
- Run `DateUtilsTest`
- Test a new order completion and verify that the Stats is updated for the current day.
